### DISCUSSION TITLE
Languages: Hack ??

### DISCUSSION THREAD

**Discussion by Burke-Knight** [github-linguist/linguist#6669] [Classification] [answered]
![GitHub_Lang](https://github.com/github-linguist/linguist/assets/121735661/7d00fc28-5f5d-407e-a125-478be10707cb)

I've noticed this in the Repo's Languages list: Hack
How can I get rid of that? The repo is PHP and CSS.

**Comment by lildude** [comment, marked as answer]
You've not provided a link to the repo where you are seeing this so I poked around your profile and I can see you've probably taken this from https://github.com/Burke-Knight/Blocks-Website-Template

The reason you are seeing this is because Hack and PHP share the `.php` extension and we use a heuristic to determine which is Hack and which is PHP:

https://github.com/github-linguist/linguist/blob/ee9119e70111d46842788b2bc1666ec1d1ea9e2a/lib/linguist/heuristics.yml#L513-L518

You have six `.php` files which don't actually contain any PHP or Hack code that match these regular expressions... in fact they don't contain any PHP code at all and are plain HTML - only `index.php` has any PHP and it is being detected correctly as PHP. As a result, things are falling through to the classifier and it's making a bad guess based on the PHP and Hack samples we have.

There are three solutions:

1. Add PHP to those files, even if it's just a `<?php // Comment ?>`, so they're really PHP files.
2. Rename the files to use `.html` extension (as that's all they contain) and use `readfile()` instead of `include()` in your `index.php` file.
3. Leave everything as-is and implement an [override](https://github.com/github-linguist/linguist/blob/master/docs/overrides.md) to force the language to PHP but be warned, you'll get PHP syntax highlighting on those HTML-only files which may not always be correct.
